src/navigation/actions.js

```diff
@@ -209,12 +217,11 @@ export const createStocktake = ({ currentUser, stocktakeName, program, itemIds }
     else if (itemIds) stocktake.setItemsByID(UIDatabase, itemIds);
   });
 
-  const stackReplacementAction = StackActions.replace({
-    routeName: 'stocktakeEditor',
-    params: { stocktake, title: navStrings.stocktake },
-  });
-
-  dispatch(stackReplacementAction);
+  // If creating a Stocktake with a program or with a list of itemIds, navigate to
+  // the StocktakeEditPage. Otherwise, navigate to the StocktakeManagePage.
+  return program || itemIds
```

Comment:
Maybe a constant here instead of the first line of the comment? E.g.

```
// If stocktake has items, navigate to StocktakeEditPage, else navigate to StocktakeManagePage.
const stocktakeHasItems = program || itemIds
return dispatch(stocktakeHasItems 
  ? gotoStocktakeEditPage(stocktake) 
  : gotoStocktakeManagePage(stocktakeName, stocktake))